zerver/decorator.py

```diff
@@ -60,25 +60,10 @@
 log_to_file(webhook_unexpected_events_logger,
             settings.WEBHOOK_UNEXPECTED_EVENTS_LOG_PATH)
 
-class _RespondAsynchronously:
-    pass
-
-# Return RespondAsynchronously from an @asynchronous view if the
-# response will be provided later by calling handler.zulip_finish(),
-# or has already been provided this way. We use this for longpolling
-# mode.
-RespondAsynchronously = _RespondAsynchronously()
-
-AsyncWrapperT = Callable[..., Union[HttpResponse, _RespondAsynchronously]]
-def asynchronous(method: Callable[..., Union[HttpResponse, _RespondAsynchronously]]) -> AsyncWrapperT:
-    # TODO: this should be the correct annotation when mypy gets fixed: type:
-    #   (Callable[[HttpRequest, base.BaseHandler, Sequence[Any], Dict[str, Any]],
-    #     Union[HttpResponse, _RespondAsynchronously]]) ->
-    #   Callable[[HttpRequest, Sequence[Any], Dict[str, Any]], Union[HttpResponse, _RespondAsynchronously]]
-    # TODO: see https://github.com/python/mypy/issues/1655
+def asynchronous(method: Callable[..., HttpResponse]) -> HttpResponse:
```

Comment:
Should be `-> Callable[..., HttpResponse]`. (Mypy doesn’t notice only because it doesn’t know what `HttpResponse` is.)

Reply:
We're deleting this function in the next commit anyway, so probably not worth fixing.